# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //third_party/libusb git repository of OpenHarmony 3.1 codebase.

Building with -Werror with both clang and gcc requires some differences in which
warnings are disabled. The pragma-pack warning in clang is not known by gcc, so we are not
allowed to disable them.

Signed-off-by: Esben Haabendal <esben.haabendal@huawei.com>
Upstream-Status: Pending

diff --git a/third_party/libusb/BUILD.gn b/third_party/libusb/BUILD.gn
index 4a4e01bbfc55..0b0d44ff7135 100644
--- a/third_party/libusb/BUILD.gn
+++ b/third_party/libusb/BUILD.gn
@@ -42,8 +42,10 @@ config("libusb_config") {
     "-Wno-#warnings",
     "-Wno-error=sign-compare",
     "-Wno-error=switch",
-    "-Wno-error=pragma-pack",
   ]
+  if (is_clang) {
+    cflags += [ "-Wno-error=pragma-pack" ]
+  }
   if (is_linux || is_ohos) {
     include_dirs += [ "linux" ]
   } else if (is_mingw || is_win) {
